src/txn/txn_recover.c

```diff
@@ -750,6 +750,15 @@ __hs_exists(WT_SESSION_IMPL *session, WT_CURSOR *metac, const char *cfg[], bool
                     WT_ERR(wt_session->salvage(wt_session, WT_HS_URI, NULL));
                 } else
                     WT_ERR(ret);
+            } else {
+                /* The statistics server is already running, make sure we don't race. */
+                WT_WRITE_BARRIER();
```

Comment:
With which write in this thread does the `F_SET` below need to be ordered?

I see that we also do a write barrier in the other place this flag is set, so I suppose it doesn't hurt to include it here if there is any doubt. But that being said, I also don't understand what the statistics thread interacts with in this thread that needs this ordering, even in the other place. If you know, I'd appreciate if you could mention it in the comment so that it's clear—if not, it's ok.

Reply:
The barrier code is no longer needed. It was required before the history store was implemented. The function `__wt_las_stats_update` function updates the statistics based on the `LAS_OPEN` flag, which gets called from the stat log server. Due to this reason, the barrier was required. Now, with durable history, the stats are updated in a different location, and it also doesn't have any interaction with this flag.